### PR TITLE
Use shared policy package artifacts across NRP tests 

### PR DESCRIPTION
**Pull request by robertschaedler3**
## Description

- Consolidate NRP test workflows to use a single NRP package artifact built on Ubuntu 14 (containing ASB/SSH ZIPs) for all tests.
- Remove duplicate workflow definitions (`github.yml`, `nrp-build.yml`)
- Fix workflow triggers (1ES pool workflow running on all `push` events)
 
## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

Validated/manually triggered:

{2C187380-EF9A-46BB-9EA9-E6EF08D84DB4}

### CODE REVIEW

**Review by AhmedBM — APPROVED** [review]
👌

**Review by simonjaeger — APPROVED** [review]

File: .github/workflows/universalnrp-test.yml [hunk 1]
```diff
@@ -7,9 +7,27 @@ on:
...
+          [
+            { os: ubuntu, version: 14.04, arch: amd64, dist: trusty, package-type: DEB },
+          ]
+    with:
+      target: ${{ matrix.target.os }}-${{ matrix.target.version }}
+      arch: ${{ matrix.target.arch }}
+      artifact: nrp-test-package
+      package-type: ${{ matrix.target.package-type }}
+      machine-config: true
+      release: ${{ github.event_name == 'pull_request' && false || true }}
```
Comment (MariusNi) [inline .github/workflows/universalnrp-test.yml:25]: Move out this to original nrp-build.yml as that one is also used to build the package as artifacts that get posted in GitHub to be either published with MC later or posted under one of our releases.

File: .github/workflows/universalnrp-test.yml [hunk 2]
```diff
@@ -7,9 +7,27 @@ on:
...
+    strategy:
+      matrix:
+        target:
+          [
+            { os: ubuntu, version: 14.04, arch: amd64, dist: trusty, package-type: DEB },
+          ]
+    with:
+      target: ${{ matrix.target.os }}-${{ matrix.target.version }}
+      arch: ${{ matrix.target.arch }}
+      artifact: nrp-test-package
```
Comment (MariusNi) [inline .github/workflows/universalnrp-test.yml:22]: >artifact: nrp-test-package [](http://example.com/codeflow?start=6&length=26)

Please do not do this. Instead, build correctly named SSH and ASB packages that will be both released as artifacts and tested here, under those exact names. 

The package names matter and the MOF inside them has that name set in.

**Review by MariusNi — DISMISSED** [review]
:clock1:

File: .github/workflows/nrp-build.yml [hunk 3]
```diff
@@ -1,45 +0,0 @@
...
-
-jobs:
-  package:
-    uses: ./.github/workflows/package-build.yml
-    with:
-      target: ${{ inputs.target }}
-      arch: ${{ inputs.arch }}
-      artifact: ${{ inputs.target }}-${{ inputs.arch }}
-      package-type: ${{ inputs.package-type }}
-      container-tag: ${{ inputs.container-tag }}
```
Comment (MariusNi) [inline .github/workflows/nrp-build.yml:43]: Add this back until we get the OneBranch build pipeline for.

File: .github/workflows/github.yml [hunk 4]
```diff
@@ -1,33 +0,0 @@
...
-    inputs:
-      ref:
-        description: 'Git ref to publish'
-        required: true
-        default: main
-        type: string
-  push:
-    # tags:
-    #   - v*.*.*
-
```
Comment (MariusNi) [inline .github/workflows/github.yml:14]: Add this file back until we get the OneBranch build pipeline up.

File: .github/workflows/package-build.yml [hunk 5]
```diff
@@ -42,7 +42,7 @@ on:
         type: string
         required: false
         default: ${{ github.ref }}
-      buildOsConfigForMc:
+      machine-config:
```
Comment (MariusNi) [inline .github/workflows/package-build.yml:45]: >machine-config: [](http://example.com/codeflow?start=6&length=15)

Please revert to previous name

File: .github/workflows/universalnrp-test.yml (hunk 2 above)
Comment (robertschaedler3) [reply .github/workflows/universalnrp-test.yml:22]: Discussed offline: The artifact name still maintains the original policy package names internally. 

\+ restored the `nrp-build.yml` workflow

**Review by MariusNi — APPROVED** [review]
:shipit:

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**